### PR TITLE
Adds ATI Tracking to Topics Page 

### LINKED ISSUE #9892: Add ATI tracking to Topic page
The current Topic pages have ATI page tracking, which is the minimum for the new pages.
Ideally, we can add chartbeat to the pages and any other ATI tracking.

**Describe the solution you'd like**
- Click and view level tracking for mobile and desktop for new Topics page via ATI
This is currently done in Morph via reverb, so these values will need to be replicated in Simorgh
- This can use the same code as the homepage but change x7 value to reflect the page type ( x7=[index-category] )

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Checklist**
- [x] (BBC contributors only) This issue follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Additional context**
- Current Topic page with tracking: https://www.bbc.com/pidgin/topics/c95y35941vrt

*Current Reverb tags:*
```
https://a1.api.bbc.co.uk/hit.xiti
?s=598342
&idclient=2704e4bc-fa8e-457c-b8c9-50652587fd5c
&ts=1643898538691
&vtag=5.29.1
&ptag=js
&r=1920x1080x24x24
&re=1920x541
&hl=14x28x58
&lng=en-GB
&idp=1428583918370
&jv=0
&p=pidgin.topics.c95y35941vrt.page
&s2=70&x2=[responsive]
&x3=[news-pidgin]
&x4=[pcm-NG]
&x5=[https%3A%2F%2Fwww.bbc.com%2Fpidgin%2Ftopics%2Fc95y35941vrt]
&x7=[index-category]
&x8=[reverb-3.3.0]
&x9=[Donald%20Trump%20-%20BBC%20News%20Pidgin]
&content_type=index-category
&language=pcm-NG
&library_version=reverb-3.3.0
&app_name=news-pidgin
&app_type=responsive&ref= 
```

*Current output from Simorgh index:*
```
https://a1.api.bbc.co.uk/hit.xiti
?s=598342
&idclient=2704e4bc-fa8e-457c-b8c9-50652587fd5c
&s2=70
&p=pidgin.page
&r=1920x1080x24x24
&re=1920x541
&hl=14x33x27
&lng=en-GB
&x1=[urn%3Abbc%3Acps%3A52f3c7b1-83ed-2338-e050-17ac8045cd94]
&x2=[responsive]
&x3=[news-pidgin]
&x4=[pcm]
&x5=[https%253A%252F%252Fwww.bbc.com%252Fpidgin]
&x7=[index-home]
&x8=[simorgh]
&x9=[Domot%2B-%2BBBC%2BNews%2BPidgin]
&x11=[2017-07-07T15%3A16%3A48.000Z]
&x12=[2022-02-03T14%3A17%3A51.000Z]
```

### PR DESCRIPTION
**Pull request by pvaliani**
Resolves #9892 

**Overall change:**
- Adds ATI tracking to topics pages - page level tracking
- Adds unit tests for topic page `url` and `params` 

x11 and x12 values were not added in this instance but may be added in the future.

**Code changes:**

As above

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/ATIAnalytics/params/index.js [hunk 1]
```diff
@@ -37,6 +41,8 @@ const pageTypeUrlBuilders = {
   mostWatched: buildMostWatchedATIUrl,
   IDX: buildIndexPageATIUrl,
   FIX: buildIndexPageATIUrl,
+  TOPIC: (data, requestContext, serviceContext) =>
+    buildTopicPageATIUrl(data, requestContext, serviceContext),
```
Comment (RetiredBlob) [inline src/app/containers/ATIAnalytics/params/index.js]: I believe this wrapping code has no effect
```suggestion
  TOPIC: buildTopicPageATIUrl,
```

File: src/app/containers/ATIAnalytics/params/index.js [hunk 2]
```diff
@@ -68,6 +74,8 @@ const pageTypeParamBuilders = {
   mostWatched: buildMostWatchedATIParams,
   IDX: buildIndexPageATIParams,
   FIX: buildIndexPageATIParams,
+  TOPIC: (data, requestContext, serviceContext) =>
+    buildTopicPageATIParams(data, requestContext, serviceContext),
```
Comment (RetiredBlob) [inline src/app/containers/ATIAnalytics/params/index.js]:
```suggestion
  TOPIC: buildTopicPageATIParams,
```

File: src/app/containers/ATIAnalytics/index.jsx [hunk 3]
```diff
@@ -24,6 +25,7 @@ const ATIAnalytics = ({ data }) => {
     mostWatched: buildMostWatchedATIUrl,
     IDX: buildIndexPageATIUrl,
     FIX: buildIndexPageATIUrl,
+    TOPIC: () => buildTopicPageATIUrl(data, requestContext, serviceContext),
```
Comment (RetiredBlob) [inline src/app/containers/ATIAnalytics/index.jsx]: Line 59 is passing these arguments anyway

```suggestion
    TOPIC: buildTopicPageATIUrl,
```

File: src/app/containers/ATIAnalytics/params/topicPage/buildParams.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+export const buildTopicPageATIUrl = (
+  topicPageData,
+  requestContext,
+  serviceContext,
+) => {
+  return buildATIPageTrackPath(
+    buildTopicPageATIParams(topicPageData, requestContext, serviceContext),
+  );
+};
```
Comment (RetiredBlob) [inline src/app/containers/ATIAnalytics/params/topicPage/buildParams.js]: Can implicit return here
```suggestion
) => buildATIPageTrackPath(
  buildTopicPageATIParams(topicPageData, requestContext, serviceContext),
);
```

**Review by RetiredBlob — APPROVED** [review]
👍  Nice!

**Review by ryanmccombe — APPROVED** [review]

File: src/app/containers/ATIAnalytics/params/topicPage/buildParams.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+export const buildTopicPageATIUrl = (
+  topicPageData,
+  requestContext,
+  serviceContext,
+) => buildATIPageTrackPath(
+  buildTopicPageATIParams(topicPageData, requestContext, serviceContext),
+);
+
+export default buildTopicPageATIParams;
```
Comment (ryanmccombe) [inline src/app/containers/ATIAnalytics/params/topicPage/buildParams.js]: Not sure this default export is actually being used - although the equivalent functions of other page types have it 😕

**Review by jroebu14 — APPROVED** [review]
Nice

File: src/app/containers/ATIAnalytics/params/topicPage/buildParams.js (hunk 5 above)
Comment (nicola-d-bbc) [reply src/app/containers/ATIAnalytics/params/topicPage/buildParams.js]: Removing it for now - if we need it in the future can just add it back in :)

**Comment by rebeccamcginn** [comment]
page: `${service}.topics.${id}.page`

**Review by rebeccamcginn — APPROVED** [review]

**Pull request merged**